contracts/pcv/maker/MakerRouter.sol

```diff
@@ -0,0 +1,170 @@
+    function swapAllFeiForDai(address to)
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        _redeemAllBalanceFromFeiPSM(to);
+    }
+
+    /// @notice Function to swap from FEI to DAI
+    /// @dev Function will swap from FEI to DAI first then DAI to USDC
+    /// @param amountFeiIn the amount of FEI to be deposited
+    /// @param minDaiAmountOut the minimum amount of DAI expected to be received
+    /// @param to the address the DAI should be sent to once swapped
+    function swapFeiForUsdc(
+        uint256 amountFeiIn,
+        uint256 minDaiAmountOut,
+        address to
+    )
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        _redeemFromFeiPSM(amountFeiIn, minDaiAmountOut, address(this));
+        daiPSM.buyGem(to, (minDaiAmountOut) / 1e12);
+    }
+
+    /// @notice Function to swap all of FEI balance to USDC
+    /// @param to the address the USDC should be sent to once swapped
+    function swapAllFeiForUsdc(address to)
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        uint256 minDaiAmountOut = _redeemAllBalanceFromFeiPSM(address(this));
+        daiPSM.buyGem(to, (minDaiAmountOut) / 1e12);
+    }
+
+    /// @notice Function to swap for both DAI and USDC
+    /// @dev Function will swap from FEI to DAI first then DAI to USDC
+    /// @param amountFeiIn the amount of FEI to be deposited
+    /// @param minDaiAmountOut the minimum amount of DAI expected to be received
+    /// @param ratioUSDC the ratio of the DAI received we would like to swap to USDC - in basis point terms
+    /// @param to the address the DAI should be sent to once swapped
+    function swapFeiForUsdcAndDai(
+        uint256 amountFeiIn,
+        uint256 minDaiAmountOut,
+        uint256 ratioUSDC,
+        address to
+    )
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        require(ratioUSDC < 10000, "MakerRouter: USDC Ratio too high");
+
+        _redeemFromFeiPSM(amountFeiIn, minDaiAmountOut, address(this));
+        uint256 usdcAmount = (minDaiAmountOut * ratioUSDC) / 10000;
+
+        daiPSM.buyGem(to, usdcAmount / 1e12);
+        dai.safeTransfer(to, minDaiAmountOut - usdcAmount);
+    }
+
+    /// @notice Function to swap all FEI balance for both DAI and USDC
+    /// @param to the address the USDC should be sent to once swapped
+    /// @param ratioUSDC the ratio of the DAI received we would like to swap to USDC - in basis point terms
+    function swapAllFeiForUsdcAndDai(address to, uint256 ratioUSDC)
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        uint256 minDaiAmountOut = _redeemAllBalanceFromFeiPSM(address(this));
+        uint256 usdcAmount = (minDaiAmountOut * ratioUSDC) / 10000;
+
+        daiPSM.buyGem(to, usdcAmount / 1e12);
```

Comment:
let's replace all instances of 1e12 with a reference to a constant variable usdcScalingFactor which equals 1e12